services/streammanager/kafka/kafkamanager.go

```diff
@@ -82,269 +139,227 @@ func loadConfig() {
+		c, err = client.New("tcp", host+":"+destConfig.Port, clientConf)
+		if err != nil {
+			continue
 		}
-		config.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA512
-	case "sha256":
-		config.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient {
-			return &xdgSCRAMClient{HashGeneratorFcn: SHA256}
+		if err = c.Ping(context.TODO()); err != nil {
+			continue
 		}
-		config.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA256
-	default:
-		return fmt.Errorf("[Kafka] invalid SASL type %s", destConfig.SaslType)
+
+		return c.NewProducer(destConfig.Topic, client.ProducerConfig{
+			WriteTimeout:  o.Timeout,
+			RetryInterval: kafkaProducerRetryInterval,
+			MaxRetries:    kafkaProducerMaxRetries,
+		})
 	}
-	return nil
-}
 
-type Opts struct {
-	Timeout time.Duration
+	return nil, fmt.Errorf("could not create producer [%q at %s]: %w", destConfig.HostName, destConfig.Port, err)
 }
 
-// NewProducerForAzureEventHub creates a producer for Azure event hub based on destination config
-func NewProducerForAzureEventHub(destinationConfig interface{}, o Opts) (sarama.SyncProducer, error) {
-	var destConfig = AzureEventHubConfig{}
+// NewProducerForAzureEventHubs creates a producer for Azure event hub based on destination config
+func NewProducerForAzureEventHubs(destinationConfig interface{}, o Opts) (*client.Producer, error) {
+	var destConfig = azureEventHubConfig{}
 	jsonConfig, err := json.Marshal(destinationConfig)
 	if err != nil {
 		return nil, fmt.Errorf(
-			"[Confluent Cloud] Error while marshaling destination Config %+v, with Error : %w",
-			destinationConfig, err)
+			"[Azure Event Hubs] Error while marshaling destination configuration %+v, got error: %w",
+			destinationConfig, err,
+		)
 	}
 	err = json.Unmarshal(jsonConfig, &destConfig)
 	if err != nil {
 		return nil, fmt.Errorf(
-			"[Confluent Cloud] Error while UnMarshaling destination Config %+v, with Error : %w",
-			destinationConfig, err)
+			"[Azure Event Hubs] Error while unmarshaling destination configuration %+v, got error: %w",
+			destinationConfig, err,
+		)
 	}
 
-	hostName := destConfig.BootstrapServer
-	hosts := []string{hostName}
-
-	conf := getDefaultConfiguration()
-
-	conf.Net.SASL.Enable = true
-	conf.Net.SASL.User = azureEventHubUser
-	conf.Net.SASL.Password = destConfig.EventHubsConnectionString
-	conf.Net.SASL.Mechanism = sarama.SASLTypePlaintext
-
-	conf.Net.TLS.Enable = true
-	conf.Net.TLS.Config = &tls.Config{ // skipcq: GO-S1020
-		InsecureSkipVerify: true,
-		ClientAuth:         0,
+	if err = destConfig.validate(); err != nil {
+		return nil, fmt.Errorf("invalid configuration: %w", err)
 	}
 
-	conf.Producer.Timeout = o.Timeout
+	c, err := client.NewAzureEventHubs(
+		destConfig.BootstrapServer, destConfig.EventHubsConnectionString, client.Config{},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("[Azure Event Hubs] Cannot create client: %w", err)
+	}
+	if err = c.Ping(context.TODO()); err != nil {
```

Comment:
no timeout here? shouldn't we use `kafkaDialTimeout` instead?

Reply:
The dial timeout is already used and injected via `client.Config` and honoured by `Ping()` but I'm going to build a context with the same timeout just in case.